dlt/helpers/airflow_helper.py

```diff
@@ -291,8 +296,28 @@ def log_after_attempt(retry_state: RetryCallState) -> None:
                         pt >> nt
                     pt = nt
                 return tasks
+            elif decompose == "parallel":
+                if not isinstance(data, DltSource):
+                    raise ValueError("Can only decompose dlt sources")
+
+                if pipeline.full_refresh:
+                    raise ValueError("Cannot decompose pipelines with full_refresh set")
+
+                # parallel tasks
+                tasks = []
+                group_name = task_name(pipeline, data)
+                start = DummyOperator(task_id=f"{group_name}_start")
+
+                with TaskGroup(group_id=group_name):
+                    for source in data.decompose("scc"):
+                        tasks.append(make_task(pipeline, source))
+
+                end = DummyOperator(task_id=f"{group_name}_end")
+
+                start >> tasks >> end
+                return [start] + tasks + [end]
```

Comment:
Not 100% sure that we should return `start` and `end` too. On the other hand - they are marking the start and the end of the parallel group. What do you think?

Reply:
hmmm so here is one trick I asked to implement:
the `start` should be the first source component. when you run the pipeline, it will do a lot of setup work (ie create schemas and initial state) that rather should not happen in parallel.

full parallelism happens in `parallel-isolated` mode

the end task makes sense. I think it is used with a condition that all predecessors succeeded. look at examples in the discussion in the original tickets

Reply:
A-ah, I think I got it about `start`. So, no dummy is needed for a start, just the first element of the decomposition, and when it's finished, other tasks are ran in parallel.

About the `end`: by default, `all_success` trigger is used, so it should already work like that:
![image](https://github.com/dlt-hub/dlt/assets/26030071/4f410640-3ef1-4d38-a663-7ab510846a99)

Reply:
hmmm cool! but what if `tasks` is an empty list? maybe you should test a pipeline with just one component (resource).

had no idea that `>>` works with lists!

Reply:
It appears it'll not work fine. The `start` and the `end` will not be up/down streamed to each other if the middle list is empty - a bit unexpected for me. Fixed.